id.go

```diff
@@ -62,32 +104,52 @@ func (id *ID) Bytes() (out []byte) {
 // Note: This is referenced as a non-pointer so it can be called directly
 // from a struct utilizing the non-pointer value of ID
 func (id *ID) String() (out string) {
+	if id == nil {
+		return
+	}
+
 	out = b64.EncodeToString(id[:])
 	return
 }
 
 // IsEmpty will return if an ID is empty
 func (id *ID) IsEmpty() (empty bool) {
+	if id == nil {
+		return true
+	}
+
+	// Iterate through each of the ID bytes
 	for i := 0; i < 16; i++ {
 		if (*id)[i] != 0 {
```

Comment:
no need to dereference array pointers, also this whole function can be replaced with:

```go
var emptyID ID
func(id *ID) IsEmpty() bool { return id != nil && *id != empttyID }
```